cmd/ilonabot.go

```diff
@@ -38,10 +43,172 @@ var ilonabotCmd = &cobra.Command{
+		return handleVersionCommand(command, client)
+	}
+
+	return nil
+}
+
+// handleVersionCommand will take care of /version submissions
+func handleVersionCommand(command slack.SlashCommand, client *slack.Client) error {
+	// The Input is found in the text field so
+	// Create the attachment and assigned based on the message
+	attachment := slack.Attachment{}
+	// Add Some default context like user who mentioned the bot
+	attachment.Fields = []slack.AttachmentField{
+		{
+			Title: "Версія:",
+			Value: appVersion,
+		},
+	}
+	// Greet the user
+
+	attachment.Pretext = fmt.Sprint("Поточна версія Slack боту IlonaBot")
+	attachment.Color = "#FF813F"
+
+	// Send the message to the channel
+	// The Channel is available in the command.ChannelID
+	_, _, err := client.PostMessage(command.ChannelID, slack.MsgOptionAttachments(attachment))
+	if err != nil {
+		return fmt.Errorf("failed to post message: %w", err)
+	}
+	return nil
+}
+
+// handleEventMessage will take an event and handle it properly based on the type of event
+func handleEventMessage(event slackevents.EventsAPIEvent, client *slack.Client) error {
+	switch event.Type {
+	// First we check if this is an CallbackEvent
+	case slackevents.CallbackEvent:
+
+		innerEvent := event.InnerEvent
+		// Yet Another Type switch on the actual Data to see if its an AppMentionEvent
+		switch ev := innerEvent.Data.(type) {
+		case *slackevents.AppMentionEvent:
+			// The application has been mentioned since this Event is a Mention event
+			err := handleAppMentionEvent(ev, client)
+			if err != nil {
+				return err
+			}
+		}
+	default:
+		return errors.New("unsupported event type")
+	}
+	return nil
+}
+
+// handleAppMentionEvent is used to take care of the AppMentionEvent when the bot is mentioned
+func handleAppMentionEvent(event *slackevents.AppMentionEvent, client *slack.Client) error {
+
+	// Grab the user name based on the ID of the one who mentioned the bot
+	user, err := client.GetUserInfo(event.User)
+	if err != nil {
+		return err
+	}
+	// Check if the user said Hello to the bot
+	text := strings.ToLower(event.Text)
+
+	// Create the attachment and assigned based on the message
+	attachment := slack.Attachment{}
+	// Add Some default context like user who mentioned the bot
+	attachment.Fields = []slack.AttachmentField{
+		{
+			Title: "Дата та час звернення",
+			Value: time.Now().Format("02-01-2006 15:04:05"),
+		}, {
+			Title: "Ініціатор звернення",
+			Value: user.Name,
+		},
+	}
+	if strings.Contains(text, "hello") || strings.Contains(text, "привіт") || strings.Contains(text, "вітаю") {
+		// Greet the user
+		// attachment.Text = "Привіт"
+		attachment.Pretext = fmt.Sprintf("Привіт %s", user.RealName)
+		attachment.Color = "#4af030"
+	} else {
+		// Send a message to the user
+		attachment.Text = fmt.Sprintf("Як я можу Вам допомогти %s?", user.RealName)
+		attachment.Pretext = "Знаходжусь в очікуванні команд"
+		attachment.Color = "#3d3d3d"
```

Comment:
Можна зробити відповіді бота менш сабмісівними
```suggestion
		attachment.Text = fmt.Sprintf("%s, слухаю уважно 😙💨", user.RealName)
		attachment.Pretext = "🥱, а? Шо, щось сталося?"
		attachment.Color = "#3d3d3d"
```